core/flyout.js

```diff
@@ -623,7 +648,52 @@ Blockly.Flyout.prototype.wheel_ = function(e) {
  * @return {boolean} True if visible.
  */
 Blockly.Flyout.prototype.isVisible = function() {
-  return this.svgGroup_ && this.svgGroup_.style.display == 'block';
+  return this.isVisible_;
+};
+
+ /**
+ * Set whether the flyout is visible.  It might be invisible because
+ * it is in a closed toolbox or because the workspace it controls
+ * is invisible.
+ * Only applies to non-paired scrollbars.
+ * @param {boolean} visible True if visible.
+ */
+Blockly.Flyout.prototype.setVisible = function(visible) {
+  var visibilityChanged = (visible != this.isVisible());
+
+  this.isVisible_ = visible;
+  if (visibilityChanged) {
+    this.updateDisplay_();
+  }
+};
+
+/**
+ * Set whether this flyout's container is visible.
+ * @param {boolean} visible Whether the container is visible.
+ */
+Blockly.Flyout.prototype.setContainerVisible = function(visible) {
+  var visibilityChanged = (visible != this.containerVisible_);
+  this.containerVisible_ = visible;
+  if (visibilityChanged) {
+    this.updateDisplay_();
+  }
+};
+
+/**
++ * Update the visibility of the flyout based whether it thinks it should
```

Comment:
How about "Update the rendering of the flyout..."
Since we're otherwise overloading "visible" a lot.